### ISSUE TITLE
gatsby .cache directory

### ISSUE THREAD

**Issue by neptunian** [gatsbyjs/gatsby#4950]
I'm new to Gatsby and am looking for information on how the .cache directory works.  I wasn't able to find any documentation.  Can someone explain or post a link to how Gatsby uses it? Do I need to commit this directory into version control?

**Comment by KyleAMathews** [comment]
Yeah, some docs on this would be great. To answer your initial question, you shouldn't commit the directory into version control.

**Comment by octalmage** [comment]
I'd also like to learn more about the .cache directory. For example, is it safe to tell TravisCI to cache this directory between builds?

**Comment by jdfm** [comment]
One issue I've run into related to the `.cache` folder (in v1 at least) is `gatsby build` vs `gatsby build --prefix-paths`. It doesn't differentiate between these cases, and so, in my case when using transformer-remark the links I was expecting to point to pathPrefix-ed versions weren't actually prefixed, and only after clearing out the `.cache` folder and running the latter did I start seeing what I was expecting.

**Comment by Rolando-Barbella** [comment]
My .cache directory despair while running updates, now if I run gatsby-develop it just breaks off course

**Comment by devrchancay** [comment]
So far I do not have problems with the `.cache` folder at the time also look for documentation about that, it would be very good that there are at least references on the website.

**Comment by tusharjoshi** [comment]
I checked the documentation and there are two references of .cache folder as follows
https://www.gatsbyjs.org/docs/gatsby-project-structure/ - mentions .cache shall be added to .gitignore

https://www.gatsbyjs.org/docs/webpack-and-ssr/#cache - explains more about the contents of .cache folder

I believe this completes the request in this issue and hence this issue can be closed.

**Comment by bealearnscode** [comment]
I just started using Gatsby today. When I ran "gatsby develop", I got two errors. The first said "Unable to copy site files to .cache". The second said "Error: EISDIR: illegal operation on a directory, readlink..." and lists the path to the .cashe directory in my project.

My very limited understanding of EISDIR tells me the .cache directory should be a single file. Is that correct? If possible, how can I make a .cache file?
